File: CODE/T4007/T4007.cpp
#define nametask "T4007"
#include <bits/stdc++.h>
#define fi first
#define se second
#define mp make_pair
#define pb push_back
#define SZ(a) (int) a.size()
#define all(a) a.begin(), a.end()
#define FOR(i, a, b) for (int i = a; i <= b; i++)
#define FOD(i, b, a) for (int i = b; i >= a; i--)

using namespace std;
typedef long long ll;
typedef pair <int, int> pi;

ll A, B;
ll f[20][200];

vector <int> getdigit(ll x){
    vector <int> res;
    while (x > 0){
        res.pb(x % 10);
        x /= 10;
    }
    return res;
}

ll calc(int id, bool tight, int sum, vector <int> digit){
    if (id < 0) {
        return (sum == 101);
    }
    if (!tight && f[id][sum] != -1) return f[id][sum];
    int k = (tight) ? digit[id] : 9;
    ll cnt = 0;
    FOR(i, 0, k){
        bool new_tight = (i == digit[id]) ? tight : 0;
        int new_sum = sum + ((id % 2 == 0) ? -i : i);
        cnt += calc(id - 1, new_tight, new_sum, digit);
    }
    if (tight) return cnt;
    return (f[id][sum] = cnt);
}

void solve(){
    memset(f, -1, sizeof f);
    vector <int> digitA, digitB;
    digitA = getdigit(A - 1);
    digitB = getdigit(B);
    cout << calc(SZ(digitB) - 1, 1, 100, digitB) - calc(SZ(digitA) - 1, 1, 100, digitA);
}

signed main()
{
    ios_base::sync_with_stdio(false); cin.tie(NULL); cout.tie(NULL);
    if (fopen(nametask".inp", "r")){
        freopen(nametask".inp", "r", stdin);
        freopen(nametask".out", "w", stdout);
    }
    cin >> A >> B;
    solve();
    return 0;
}

File: CODE/T4007/T4007.py
import sys

sys.stdin = open("T4007.inp", "r")
sys.stdout = open("T4007.out", "w")

def get_digit(x):
    v = []
    while (x > 0):
        v.append(x % 10)
        x //= 10
    return v

def digit_dp(idx, tight, total, digits, dp):
    if idx == -1:
        return 1 if total == 101 else 0
    if not tight and dp[idx][total] != -1:
        return dp[idx][total]
    
    max_digit = digits[idx] if tight else 9
    cnt = 0
    
    for i in range(max_digit + 1):
        new_tight = tight and (i == digits[idx])
        new_total = total + (i if idx % 2 else -i)
        cnt += digit_dp(idx - 1, new_tight, new_total, digits, dp)
    
    if not tight:
        dp[idx][total] = cnt
    return cnt

def count_special_numbers(A, B):
    dp = [[-1] * 200 for _ in range(20)]
    digitsB = get_digit(B)
    digitsA = get_digit(A - 1)
    return digit_dp(len(digitsB) - 1, True, 100, digitsB, dp) - digit_dp(len(digitsA) - 1, True, 100, digitsA, dp)

A, B = map(int, input().split())
print(count_special_numbers(A, B))
